src/dappsExplorer/DiscoverDappsCard.tsx

```diff
@@ -0,0 +1,195 @@
+import useOpenDapp from 'src/dappsExplorer/useOpenDapp'
+import { currentLanguageSelector } from 'src/i18n/selectors'
+import { Screens } from 'src/navigator/Screens'
+import { useDispatch, useSelector } from 'src/redux/hooks'
+import { Colors } from 'src/styles/colors'
+import fontStyles, { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import TextButton from 'src/components/TextButton'
+import { navigate } from 'src/navigator/NavigationService'
+
+const AnimatedSectionList =
+  Animated.createAnimatedComponent<SectionListProps<Dapp, SectionData>>(SectionList)
+
+interface SectionData {
+  data: Dapp[]
+  sectionName: string
+  dappSection: DappSection
+  testID: string
+}
+
+const MAX_DAPPS = 5
+
+function DiscoverDappsCard() {
+  const { t } = useTranslation()
+
+  const insets = useSafeAreaInsets()
+
+  const sectionListRef = useRef<SectionList>(null)
+
+  const language = useSelector(currentLanguageSelector)
+  const dispatch = useDispatch()
+  const loading = useSelector(dappsListLoadingSelector)
+  const favoriteDapps = useSelector(favoriteDappsSelector)
+  const mostPopularDapps = useSelector(mostPopularDappsSelector)
+
+  const { onSelectDapp } = useOpenDapp()
+
+  useEffect(() => {
+    dispatch(fetchDappsList())
+    ValoraAnalytics.track(DappExplorerEvents.dapp_screen_open)
+  }, [])
+
+  const onPressDapp = (dapp: ActiveDapp, index: number) => {
+    onSelectDapp(dapp, {
+      position: 1 + index,
+      fromScreen: Screens.TabDiscover,
+    })
+  }
+
+  const sections: SectionData[] = useMemo(() => {
+    const favoriteDappIds = favoriteDapps.map((dapp) => dapp.id)
+    let mostPopularSection: SectionData[] = []
+    if (favoriteDapps.length <= 2) {
+      mostPopularSection = [
+        {
+          data: mostPopularDapps
+            .filter((dapp) => !favoriteDappIds.includes(dapp.id))
+            .slice(0, MAX_DAPPS - favoriteDapps.length),
+          sectionName: t('dappsScreen.mostPopularDapps').toLocaleUpperCase(language ?? 'en-US'),
+          dappSection: DappSection.MostPopular,
+          testID: 'DiscoverDappsCard/MostPopularSection',
+        },
+      ]
+    }
+    const favoritesSection =
+      favoriteDapps.length > 0
+        ? [
+            {
+              data: favoriteDapps.slice(0, MAX_DAPPS),
+              sectionName: t('dappsScreen.favoriteDapps').toLocaleUpperCase(language ?? 'en-US'),
+              dappSection: DappSection.FavoritesDappScreen,
+              testID: 'DiscoverDappsCard/FavoritesSection',
+            },
+          ]
+        : []
+
+    return [...favoritesSection, ...mostPopularSection]
+  }, [favoriteDapps, mostPopularDapps])
+
+  const onPressExploreAll = () => {
+    ValoraAnalytics.track(DappExplorerEvents.dapp_explore_all)
+    navigate(Screens.DappScreen)
+  }
+
+  if (!sections.length) return null
+
+  return (
+    <View testID="DiscoverDappsCard" style={styles.safeAreaContainer}>
+      <AnimatedSectionList
```

Comment:
Do we still need a scrollable list here?

Reply:
Do you mean just using a plan `SectionList` instead?

Reply:
It helps to deal with the section headers, since it's not guaranteed that both sections will always be present. I kind of like keeping it since it's the same logic we use elsewhere for maintaining lists with distinct sections.